### PR TITLE
chore(dependencies): Upgrade org.testng:testng to resolve vulnerability

### PR DESCRIPTION
**Pull request by j-sandy**
SONATYPE-2019-0115
org.testng:testng is transitively introduced by org.codehaus.groovy:groovy-testng (part of groovy-all)

### CODE REVIEW

**Comment by j-sandy** [comment]
After applying the fix, gate dependency insight
```
$.\gradlew gate-web:dI --dependency org.testng:testng --configuration runtimeClasspath
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details

> Task :gate-web:dependencyInsight
org.testng:testng:7.4.0
   variant "runtimeElements" [[21s]
      org.gradle.category            = library
      org.gradle.dependency.bundling = external
      org.gradle.jvm.version         = 8 (compatible with: 11)
      org.gradle.libraryelements     = jar
      org.gradle.usage               = java-runtime
      org.gradle.status              = release (not requested)
   ]
   Selection reasons:
      - By constraint
      - By conflict resolution : between versions 7.4.0 and 6.13.1

org.testng:testng:7.4.0
\--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT
     +--- runtimeClasspath
     +--- project :gate-proxy
     |    \--- runtimeClasspath
     +--- project :gate-plugins
     |    \--- runtimeClasspath
     +--- project :gate-api
     |    +--- runtimeClasspath
     |    +--- project :gate-proxy (*)
     |    \--- project :gate-plugins (*)
     +--- project :gate-integrations-gremlin
     |    \--- runtimeClasspath
     +--- project :gate-basic
     |    \--- runtimeClasspath
     +--- project :gate-iap
     |    \--- runtimeClasspath
     +--- project :gate-ldap
     |    \--- runtimeClasspath
     +--- project :gate-oauth2
     |    \--- runtimeClasspath
     +--- project :gate-saml
     |    \--- runtimeClasspath
     +--- project :gate-x509
     |    \--- runtimeClasspath
     \--- project :gate-core
          +--- runtimeClasspath
          +--- project :gate-proxy (*)
          +--- project :gate-plugins (*)
          +--- project :gate-integrations-gremlin (*)
          +--- project :gate-basic (*)
          +--- project :gate-iap (*)
          +--- project :gate-ldap (*)
          +--- project :gate-oauth2 (*)
          +--- project :gate-saml (*)
          \--- project :gate-x509 (*)

org.testng:testng:6.13.1 -> 7.4.0
\--- org.codehaus.groovy:groovy-testng:2.5.11
     +--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT (requested org.codehaus.groovy:groovy-testng:2.5.9)
     |    +--- runtimeClasspath
     |    +--- project :gate-proxy
     |    |    \--- runtimeClasspath
     |    +--- project :gate-plugins
     |    |    \--- runtimeClasspath
     |    +--- project :gate-api
     |    |    +--- runtimeClasspath
     |    |    +--- project :gate-proxy (*)
     |    |    \--- project :gate-plugins (*)
     |    +--- project :gate-integrations-gremlin
     |    |    \--- runtimeClasspath
     |    +--- project :gate-basic
     |    |    \--- runtimeClasspath
     |    +--- project :gate-iap
     |    |    \--- runtimeClasspath
     |    +--- project :gate-ldap
     |    |    \--- runtimeClasspath
     |    +--- project :gate-oauth2
     |    |    \--- runtimeClasspath
     |    +--- project :gate-saml
     |    |    \--- runtimeClasspath
     |    +--- project :gate-x509
     |    |    \--- runtimeClasspath
     |    \--- project :gate-core
     |         +--- runtimeClasspath
     |         +--- project :gate-proxy (*)
     |         +--- project :gate-plugins (*)
     |         +--- project :gate-integrations-gremlin (*)
     |         +--- project :gate-basic (*)
     |         +--- project :gate-iap (*)
     |         +--- project :gate-ldap (*)
     |         +--- project :gate-oauth2 (*)
     |         +--- project :gate-saml (*)
     |         \--- project :gate-x509 (*)
     \--- org.codehaus.groovy:groovy-all:2.5.11
          +--- runtimeClasspath (requested org.codehaus.groovy:groovy-all)
          +--- project :gate-proxy (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :gate-plugins (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :gate-integrations-gremlin (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :gate-basic (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :gate-iap (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :gate-ldap (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :gate-oauth2 (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :gate-saml (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :gate-x509 (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :gate-core (requested org.codehaus.groovy:groovy-all) (*)
          +--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT (*)
          \--- io.spinnaker.kork:kork-web:7.109.0
               +--- runtimeClasspath (requested io.spinnaker.kork:kork-web)
               +--- project :gate-proxy (requested io.spinnaker.kork:kork-web) (*)
               +--- project :gate-plugins (requested io.spinnaker.kork:kork-web) (*)
               +--- project :gate-integrations-gremlin (requested io.spinnaker.kork:kork-web) (*)
               +--- project :gate-core (requested io.spinnaker.kork:kork-web) (*)
               +--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT (requested io.spinnaker.kork:kork-web:testng-cve-fix-SNAPSHOT) (*)
               +--- io.spinnaker.fiat:fiat-api:1.28.0
               |    +--- runtimeClasspath
               |    +--- project :gate-plugins (*)
               |    +--- project :gate-iap (*)
               |    +--- project :gate-oauth2 (*)
               |    +--- project :gate-saml (*)
               |    +--- project :gate-x509 (*)
               |    \--- project :gate-core (*)
               +--- io.spinnaker.kork:kork-plugins:testng-cve-fix-SNAPSHOT (requested io.spinnaker.kork:kork-web:testng-cve-fix-SNAPSHOT)
               |    +--- runtimeClasspath (requested io.spinnaker.kork:kork-plugins)
               |    +--- project :gate-plugins (requested io.spinnaker.kork:kork-plugins) (*)
               |    +--- project :gate-core (requested io.spinnaker.kork:kork-plugins) (*)
               |    \--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT (*)
               +--- io.spinnaker.kork:kork-runtime:testng-cve-fix-SNAPSHOT (requested io.spinnaker.kork:kork-web:testng-cve-fix-SNAPSHOT)
               |    +--- runtimeClasspath (requested io.spinnaker.kork:kork-runtime)
               |    \--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT (*)
               \--- io.spinnaker.kork:kork-retrofit:testng-cve-fix-SNAPSHOT (requested io.spinnaker.kork:kork-web:testng-cve-fix-SNAPSHOT)
                    +--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT (*)
                    \--- io.spinnaker.kork:kork-runtime:testng-cve-fix-SNAPSHOT (*)
```

**Comment by j-sandy** [comment]
Clouddriver dependency insight:
```
$ .\gradlew clouddriver-web:dI --dependency org.testng:testng --configuration runtimeClasspath

> Task :clouddriver-web:dependencyInsight
org.testng:testng:7.4.0
   variant "runtimeElements" [
      org.gradle.category                = library
      org.gradle.dependency.bundling     = external
      org.gradle.jvm.version             = 8 (compatible with: 11)
      org.gradle.libraryelements         = jar
      org.gradle.usage                   = java-runtime
      org.gradle.status                  = release (not requested)

      Requested attributes not found in the selected variant:
         org.jetbrains.kotlin.platform.type = jvm
   ]
   Selection reasons:
      - By constraint
      - By conflict resolution : between versions 7.4.0 and 6.13.1

org.testng:testng:7.4.0
\--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT
     +--- runtimeClasspath
     +--- project :clouddriver-kubernetes
     |    \--- runtimeClasspath
     +--- project :clouddriver-ecs
     |    \--- runtimeClasspath
     +--- project :clouddriver-lambda
     |    \--- runtimeClasspath
     +--- project :clouddriver-appengine
     |    \--- runtimeClasspath
     +--- project :clouddriver-cloudfoundry
     |    \--- runtimeClasspath
     +--- project :clouddriver-google
     |    \--- runtimeClasspath
     +--- project :clouddriver-artifacts
     |    +--- runtimeClasspath
     |    +--- project :clouddriver-kubernetes (*)
     |    +--- project :clouddriver-ecs (*)
     |    +--- project :clouddriver-lambda (*)
     |    +--- project :clouddriver-appengine (*)
     |    +--- project :clouddriver-cloudfoundry (*)
     |    \--- project :clouddriver-google (*)
     +--- project :clouddriver-elasticsearch
     |    \--- runtimeClasspath
     +--- project :clouddriver-sql-mysql
     |    \--- runtimeClasspath
     +--- project :clouddriver-sql-postgres
     |    \--- runtimeClasspath
     +--- project :cats:cats-sql
     |    +--- project :clouddriver-sql-mysql (*)
     |    \--- project :clouddriver-sql-postgres (*)
     +--- project :clouddriver-sql
     |    +--- runtimeClasspath
     |    +--- project :clouddriver-sql-mysql (*)
     |    +--- project :clouddriver-sql-postgres (*)
     |    \--- project :cats:cats-sql (*)
     +--- project :clouddriver-tencentcloud
     |    \--- runtimeClasspath
     +--- project :clouddriver-titus
     |    \--- runtimeClasspath
     +--- project :clouddriver-aws
     |    +--- runtimeClasspath
     |    +--- project :clouddriver-ecs (*)
     |    +--- project :clouddriver-lambda (*)
     |    \--- project :clouddriver-titus (*)
     +--- project :clouddriver-eureka
     |    +--- runtimeClasspath
     |    +--- project :clouddriver-tencentcloud (*)
     |    +--- project :clouddriver-titus (*)
     |    \--- project :clouddriver-aws (*)
     +--- project :clouddriver-oracle
     |    \--- runtimeClasspath
     +--- project :clouddriver-azure
     |    \--- runtimeClasspath
     +--- project :clouddriver-consul
     |    +--- runtimeClasspath
     |    \--- project :clouddriver-google (*)
     +--- project :clouddriver-huaweicloud
     |    \--- runtimeClasspath
     +--- project :clouddriver-yandex
     |    \--- runtimeClasspath
     +--- project :clouddriver-docker
     |    +--- runtimeClasspath
     |    \--- project :clouddriver-cloudfoundry (*)
     +--- project :clouddriver-core
     |    +--- runtimeClasspath
     |    +--- project :clouddriver-kubernetes (*)
     |    +--- project :clouddriver-ecs (*)
     |    +--- project :clouddriver-lambda (*)
     |    +--- project :clouddriver-appengine (*)
     |    +--- project :clouddriver-cloudfoundry (*)
     |    +--- project :clouddriver-google (*)
     |    +--- project :clouddriver-artifacts (*)
     |    +--- project :clouddriver-elasticsearch (*)
     |    +--- project :cats:cats-sql (*)
     |    +--- project :clouddriver-sql (*)
     |    +--- project :clouddriver-tencentcloud (*)
     |    +--- project :clouddriver-titus (*)
     |    +--- project :clouddriver-aws (*)
     |    +--- project :clouddriver-eureka (*)
     |    +--- project :clouddriver-oracle (*)
     |    +--- project :clouddriver-azure (*)
     |    +--- project :clouddriver-consul (*)
     |    +--- project :clouddriver-huaweicloud (*)
     |    +--- project :clouddriver-yandex (*)
     |    \--- project :clouddriver-docker (*)
     +--- project :clouddriver-security
     |    +--- runtimeClasspath
     |    +--- project :clouddriver-kubernetes (*)
     |    +--- project :clouddriver-ecs (*)
     |    +--- project :clouddriver-lambda (*)
     |    +--- project :clouddriver-appengine (*)
     |    +--- project :clouddriver-cloudfoundry (*)
     |    +--- project :clouddriver-google (*)
     |    +--- project :clouddriver-elasticsearch (*)
     |    +--- project :cats:cats-sql (*)
     |    +--- project :clouddriver-tencentcloud (*)
     |    +--- project :clouddriver-titus (*)
     |    +--- project :clouddriver-aws (*)
     |    +--- project :clouddriver-oracle (*)
     |    +--- project :clouddriver-azure (*)
     |    +--- project :clouddriver-huaweicloud (*)
     |    +--- project :clouddriver-yandex (*)
     |    +--- project :clouddriver-docker (*)
     |    \--- project :clouddriver-core (*)
     +--- project :cats:cats-redis
     |    +--- project :cats:cats-sql (*)
     |    \--- project :clouddriver-core (*)
     +--- project :cats:cats-core
     |    +--- runtimeClasspath
     |    +--- project :clouddriver-kubernetes (*)
     |    +--- project :clouddriver-ecs (*)
     |    +--- project :clouddriver-lambda (*)
     |    +--- project :clouddriver-appengine (*)
     |    +--- project :clouddriver-cloudfoundry (*)
     |    +--- project :clouddriver-google (*)
     |    +--- project :cats:cats-sql (*)
     |    +--- project :clouddriver-sql (*)
     |    +--- project :clouddriver-tencentcloud (*)
     |    +--- project :clouddriver-titus (*)
     |    +--- project :clouddriver-aws (*)
     |    +--- project :clouddriver-eureka (*)
     |    +--- project :clouddriver-oracle (*)
     |    +--- project :clouddriver-azure (*)
     |    +--- project :clouddriver-huaweicloud (*)
     |    +--- project :clouddriver-yandex (*)
     |    +--- project :clouddriver-docker (*)
     |    +--- project :clouddriver-core (*)
     |    +--- project :clouddriver-security (*)
     |    \--- project :cats:cats-redis (*)
     +--- project :clouddriver-api
     |    +--- runtimeClasspath
     |    +--- project :clouddriver-kubernetes (*)
     |    +--- project :clouddriver-ecs (*)
     |    +--- project :clouddriver-lambda (*)
     |    +--- project :clouddriver-appengine (*)
     |    +--- project :clouddriver-cloudfoundry (*)
     |    +--- project :clouddriver-google (*)
     |    +--- project :clouddriver-artifacts (*)
     |    +--- project :clouddriver-elasticsearch (*)
     |    +--- project :cats:cats-sql (*)
     |    +--- project :clouddriver-sql (*)
     |    +--- project :clouddriver-tencentcloud (*)
     |    +--- project :clouddriver-titus (*)
     |    +--- project :clouddriver-aws (*)
     |    +--- project :clouddriver-eureka (*)
     |    +--- project :clouddriver-oracle (*)
     |    +--- project :clouddriver-azure (*)
     |    +--- project :clouddriver-huaweicloud (*)
     |    +--- project :clouddriver-yandex (*)
     |    +--- project :clouddriver-docker (*)
     |    +--- project :clouddriver-core (*)
     |    +--- project :clouddriver-security (*)
     |    +--- project :cats:cats-redis (*)
     |    \--- project :cats:cats-core (*)
     +--- project :clouddriver-google-common
     |    +--- runtimeClasspath
     |    +--- project :clouddriver-appengine (*)
     |    \--- project :clouddriver-google (*)
     +--- project :clouddriver-configserver
     |    +--- runtimeClasspath
     |    +--- project :clouddriver-kubernetes (*)
     |    \--- project :clouddriver-aws (*)
     +--- project :clouddriver-saga
     |    +--- project :clouddriver-titus (*)
     |    +--- project :clouddriver-aws (*)
     |    \--- project :clouddriver-core (*)
     \--- project :clouddriver-event
          +--- project :clouddriver-sql (*)
          \--- project :clouddriver-saga (*)

org.testng:testng:6.13.1 -> 7.4.0
\--- org.codehaus.groovy:groovy-testng:2.5.11
     +--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT (requested org.codehaus.groovy:groovy-testng:2.5.9)
     |    +--- runtimeClasspath
     |    +--- project :clouddriver-kubernetes
     |    |    \--- runtimeClasspath
     |    +--- project :clouddriver-ecs
     |    |    \--- runtimeClasspath
     |    +--- project :clouddriver-lambda
     |    |    \--- runtimeClasspath
     |    +--- project :clouddriver-appengine
     |    |    \--- runtimeClasspath
     |    +--- project :clouddriver-cloudfoundry
     |    |    \--- runtimeClasspath
     |    +--- project :clouddriver-google
     |    |    \--- runtimeClasspath
     |    +--- project :clouddriver-artifacts
     |    |    +--- runtimeClasspath
     |    |    +--- project :clouddriver-kubernetes (*)
     |    |    +--- project :clouddriver-ecs (*)
     |    |    +--- project :clouddriver-lambda (*)
     |    |    +--- project :clouddriver-appengine (*)
     |    |    +--- project :clouddriver-cloudfoundry (*)
     |    |    \--- project :clouddriver-google (*)
     |    +--- project :clouddriver-elasticsearch
     |    |    \--- runtimeClasspath
     |    +--- project :clouddriver-sql-mysql
     |    |    \--- runtimeClasspath
     |    +--- project :clouddriver-sql-postgres
     |    |    \--- runtimeClasspath
     |    +--- project :cats:cats-sql
     |    |    +--- project :clouddriver-sql-mysql (*)
     |    |    \--- project :clouddriver-sql-postgres (*)
     |    +--- project :clouddriver-sql
     |    |    +--- runtimeClasspath
     |    |    +--- project :clouddriver-sql-mysql (*)
     |    |    +--- project :clouddriver-sql-postgres (*)
     |    |    \--- project :cats:cats-sql (*)
     |    +--- project :clouddriver-tencentcloud
     |    |    \--- runtimeClasspath
     |    +--- project :clouddriver-titus
     |    |    \--- runtimeClasspath
     |    +--- project :clouddriver-aws
     |    |    +--- runtimeClasspath
     |    |    +--- project :clouddriver-ecs (*)
     |    |    +--- project :clouddriver-lambda (*)
     |    |    \--- project :clouddriver-titus (*)
     |    +--- project :clouddriver-eureka
     |    |    +--- runtimeClasspath
     |    |    +--- project :clouddriver-tencentcloud (*)
     |    |    +--- project :clouddriver-titus (*)
     |    |    \--- project :clouddriver-aws (*)
     |    +--- project :clouddriver-oracle
     |    |    \--- runtimeClasspath
     |    +--- project :clouddriver-azure
     |    |    \--- runtimeClasspath
     |    +--- project :clouddriver-consul
     |    |    +--- runtimeClasspath
     |    |    \--- project :clouddriver-google (*)
     |    +--- project :clouddriver-huaweicloud
     |    |    \--- runtimeClasspath
     |    +--- project :clouddriver-yandex
     |    |    \--- runtimeClasspath
     |    +--- project :clouddriver-docker
     |    |    +--- runtimeClasspath
     |    |    \--- project :clouddriver-cloudfoundry (*)
     |    +--- project :clouddriver-core
     |    |    +--- runtimeClasspath
     |    |    +--- project :clouddriver-kubernetes (*)
     |    |    +--- project :clouddriver-ecs (*)
     |    |    +--- project :clouddriver-lambda (*)
     |    |    +--- project :clouddriver-appengine (*)
     |    |    +--- project :clouddriver-cloudfoundry (*)
     |    |    +--- project :clouddriver-google (*)
     |    |    +--- project :clouddriver-artifacts (*)
     |    |    +--- project :clouddriver-elasticsearch (*)
     |    |    +--- project :cats:cats-sql (*)
     |    |    +--- project :clouddriver-sql (*)
     |    |    +--- project :clouddriver-tencentcloud (*)
     |    |    +--- project :clouddriver-titus (*)
     |    |    +--- project :clouddriver-aws (*)
     |    |    +--- project :clouddriver-eureka (*)
     |    |    +--- project :clouddriver-oracle (*)
     |    |    +--- project :clouddriver-azure (*)
     |    |    +--- project :clouddriver-consul (*)
     |    |    +--- project :clouddriver-huaweicloud (*)
     |    |    +--- project :clouddriver-yandex (*)
     |    |    \--- project :clouddriver-docker (*)
     |    +--- project :clouddriver-security
     |    |    +--- runtimeClasspath
     |    |    +--- project :clouddriver-kubernetes (*)
     |    |    +--- project :clouddriver-ecs (*)
     |    |    +--- project :clouddriver-lambda (*)
     |    |    +--- project :clouddriver-appengine (*)
     |    |    +--- project :clouddriver-cloudfoundry (*)
     |    |    +--- project :clouddriver-google (*)
     |    |    +--- project :clouddriver-elasticsearch (*)
     |    |    +--- project :cats:cats-sql (*)
     |    |    +--- project :clouddriver-tencentcloud (*)
     |    |    +--- project :clouddriver-titus (*)
     |    |    +--- project :clouddriver-aws (*)
     |    |    +--- project :clouddriver-oracle (*)
     |    |    +--- project :clouddriver-azure (*)
     |    |    +--- project :clouddriver-huaweicloud (*)
     |    |    +--- project :clouddriver-yandex (*)
     |    |    +--- project :clouddriver-docker (*)
     |    |    \--- project :clouddriver-core (*)
     |    +--- project :cats:cats-redis
     |    |    +--- project :cats:cats-sql (*)
     |    |    \--- project :clouddriver-core (*)
     |    +--- project :cats:cats-core
     |    |    +--- runtimeClasspath
     |    |    +--- project :clouddriver-kubernetes (*)
     |    |    +--- project :clouddriver-ecs (*)
     |    |    +--- project :clouddriver-lambda (*)
     |    |    +--- project :clouddriver-appengine (*)
     |    |    +--- project :clouddriver-cloudfoundry (*)
     |    |    +--- project :clouddriver-google (*)
     |    |    +--- project :cats:cats-sql (*)
     |    |    +--- project :clouddriver-sql (*)
     |    |    +--- project :clouddriver-tencentcloud (*)
     |    |    +--- project :clouddriver-titus (*)
     |    |    +--- project :clouddriver-aws (*)
     |    |    +--- project :clouddriver-eureka (*)
     |    |    +--- project :clouddriver-oracle (*)
     |    |    +--- project :clouddriver-azure (*)
     |    |    +--- project :clouddriver-huaweicloud (*)
     |    |    +--- project :clouddriver-yandex (*)
     |    |    +--- project :clouddriver-docker (*)
     |    |    +--- project :clouddriver-core (*)
     |    |    +--- project :clouddriver-security (*)
     |    |    \--- project :cats:cats-redis (*)
     |    +--- project :clouddriver-api
     |    |    +--- runtimeClasspath
     |    |    +--- project :clouddriver-kubernetes (*)
     |    |    +--- project :clouddriver-ecs (*)
     |    |    +--- project :clouddriver-lambda (*)
     |    |    +--- project :clouddriver-appengine (*)
     |    |    +--- project :clouddriver-cloudfoundry (*)
     |    |    +--- project :clouddriver-google (*)
     |    |    +--- project :clouddriver-artifacts (*)
     |    |    +--- project :clouddriver-elasticsearch (*)
     |    |    +--- project :cats:cats-sql (*)
     |    |    +--- project :clouddriver-sql (*)
     |    |    +--- project :clouddriver-tencentcloud (*)
     |    |    +--- project :clouddriver-titus (*)
     |    |    +--- project :clouddriver-aws (*)
     |    |    +--- project :clouddriver-eureka (*)
     |    |    +--- project :clouddriver-oracle (*)
     |    |    +--- project :clouddriver-azure (*)
     |    |    +--- project :clouddriver-huaweicloud (*)
     |    |    +--- project :clouddriver-yandex (*)
     |    |    +--- project :clouddriver-docker (*)
     |    |    +--- project :clouddriver-core (*)
     |    |    +--- project :clouddriver-security (*)
     |    |    +--- project :cats:cats-redis (*)
     |    |    \--- project :cats:cats-core (*)
     |    +--- project :clouddriver-google-common
     |    |    +--- runtimeClasspath
     |    |    +--- project :clouddriver-appengine (*)
     |    |    \--- project :clouddriver-google (*)
     |    +--- project :clouddriver-configserver
     |    |    +--- runtimeClasspath
     |    |    +--- project :clouddriver-kubernetes (*)
     |    |    \--- project :clouddriver-aws (*)
     |    +--- project :clouddriver-saga
     |    |    +--- project :clouddriver-titus (*)
     |    |    +--- project :clouddriver-aws (*)
     |    |    \--- project :clouddriver-core (*)
     |    \--- project :clouddriver-event
     |         +--- project :clouddriver-sql (*)
     |         \--- project :clouddriver-saga (*)
     \--- org.codehaus.groovy:groovy-all:2.5.11
          +--- runtimeClasspath (requested org.codehaus.groovy:groovy-all)
          +--- project :clouddriver-kubernetes (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-ecs (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-lambda (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-appengine (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-cloudfoundry (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-google (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-artifacts (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-elasticsearch (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :cats:cats-sql (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-titus (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-aws (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-eureka (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-oracle (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-azure (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-consul (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-huaweicloud (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-yandex (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-docker (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-core (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-security (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :cats:cats-redis (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :cats:cats-core (requested org.codehaus.groovy:groovy-all) (*)
          +--- project :clouddriver-google-common (requested org.codehaus.groovy:groovy-all) (*)
          +--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT (*)
          +--- io.spinnaker.kork:kork-web:7.109.0
          |    +--- runtimeClasspath (requested io.spinnaker.kork:kork-web)
          |    +--- project :clouddriver-eureka (requested io.spinnaker.kork:kork-web) (*)
          |    +--- project :clouddriver-core (requested io.spinnaker.kork:kork-web) (*)
          |    +--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT (requested io.spinnaker.kork:kork-web:testng-cve-fix-SNAPSHOT) (*)
          |    +--- io.spinnaker.kork:kork-runtime:testng-cve-fix-SNAPSHOT (requested io.spinnaker.kork:kork-web:testng-cve-fix-SNAPSHOT)
          |    |    +--- runtimeClasspath (requested io.spinnaker.kork:kork-runtime)
          |    |    \--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT (*)
          |    +--- io.spinnaker.kork:kork-retrofit:testng-cve-fix-SNAPSHOT (requested io.spinnaker.kork:kork-web:testng-cve-fix-SNAPSHOT)
          |    |    +--- project :clouddriver-aws (requested io.spinnaker.kork:kork-retrofit) (*)
          |    |    +--- project :clouddriver-core (requested io.spinnaker.kork:kork-retrofit) (*)
          |    |    +--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT (*)
          |    |    \--- io.spinnaker.kork:kork-runtime:testng-cve-fix-SNAPSHOT (*)
          |    +--- io.spinnaker.kork:kork-plugins:testng-cve-fix-SNAPSHOT (requested io.spinnaker.kork:kork-web:testng-cve-fix-SNAPSHOT)
          |    |    +--- runtimeClasspath (requested io.spinnaker.kork:kork-plugins)
          |    |    +--- project :clouddriver-core (requested io.spinnaker.kork:kork-plugins) (*)
          |    |    \--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT (*)
          |    \--- io.spinnaker.fiat:fiat-api:1.28.0
          |         +--- runtimeClasspath
          |         +--- project :clouddriver-ecs (*)
          |         +--- project :clouddriver-appengine (*)
          |         +--- project :clouddriver-google (*)
          |         +--- project :cats:cats-sql (*)
          |         +--- project :clouddriver-tencentcloud (*)
          |         +--- project :clouddriver-titus (*)
          |         +--- project :clouddriver-aws (*)
          |         +--- project :clouddriver-oracle (*)
          |         +--- project :clouddriver-azure (*)
          |         +--- project :clouddriver-huaweicloud (*)
          |         +--- project :clouddriver-docker (*)
          |         +--- project :clouddriver-core (*)
          |         +--- project :clouddriver-security (*)
          |         \--- project :clouddriver-google-common (*)
          \--- io.spinnaker.kork:kork-aws:testng-cve-fix-SNAPSHOT
               +--- project :clouddriver-aws (requested io.spinnaker.kork:kork-aws) (*)
               \--- io.spinnaker.kork:kork-bom:testng-cve-fix-SNAPSHOT (*)
```

File: spinnaker-dependencies/spinnaker-dependencies.gradle [hunk 1]
```diff
@@ -168,6 +168,7 @@ dependencies {
     api("org.springframework.security.oauth.boot:spring-security-oauth2-autoconfigure:2.1.5.RELEASE")
     api("org.springframework.security.extensions:spring-security-saml-dsl-core:1.0.5.RELEASE")
     api("org.springframework.security.extensions:spring-security-saml2-core:1.0.9.RELEASE")
+    api("org.testng:testng:7.4.0") // TODO: remove this with upgrade of spring-boot version to 2.5.0 or with upgrade of groovy-all to 3.0.8
```
Comment (dbyron-sf) [inline spinnaker-dependencies/spinnaker-dependencies.gradle:171]: As background, [groovy 3.0.7](https://github.com/apache/groovy/blob/GROOVY_3_0_7/subprojects/groovy-testng/build.gradle#L21) brings in testng 7.3.0 and [groovy 3.0.8](https://github.com/apache/groovy/blob/GROOVY_3_0_8/subprojects/groovy-testng/build.gradle#L21) brings in 7.4.0 via [this commit].(https://github.com/apache/groovy/commit/c187b719937a533b32e33a3ba5b0c035aabe3d11).

[spring boot 2.5.0](https://docs.spring.io/spring-boot/docs/2.5.0/reference/htmlsingle/#dependency-versions) brings in groovy-testng 3.0.8, where [spring boot 2.4.9](https://docs.spring.io/spring-boot/docs/2.4.9/reference/htmlsingle/#dependency-versions) brings in groovy-testng 2.5.14.  It's in theory possible that some later 2.4.x could move to groovy-testng 3.0.8, but given this [2.5 release note](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.5-Release-Notes#groovy-3), that seems unlikely.

**Comment by dbyron-sf** [comment]
I believe https://snyk.io/vuln/SNYK-JAVA-ORGTESTNG-174823 motivates updating testng to at least 7.0.0 and https://github.com/cbeust/testng/issues/2406 motivates going to 7.4.0

File: spinnaker-dependencies/spinnaker-dependencies.gradle (hunk 1 above)
Comment (dbyron-sf) [reply spinnaker-dependencies/spinnaker-dependencies.gradle:171]: Also, before this change, gate-web brought in `org.testng:testng:6.13.1` via:  
```
|    +--- org.codehaus.groovy:groovy-testng:2.5.11
|    |    +--- org.codehaus.groovy:groovy:2.5.11
|    |    \--- org.testng:testng:6.13.1
|    |         \--- com.beust:jcommander:1.72
```

**Comment by dbyron-sf** [comment]
I'm a little nervous about such a big jump in testng (from 6.13.1 to 7.4.0), but given that I only see fixes, new features, and removal of deprecated functions in https://github.com/cbeust/testng/blob/7.4.0/CHANGES.txt (i.e. nothing obviously breaking) and https://github.com/cbeust/testng/blob/7.4.0/build.gradle.kts#L108 show that testng tests against groovy 2.4 (i.e. not 3.x), I'm OK with this.

**Review by dbyron-sf — APPROVED** [review]

**Comment by j-sandy** [comment]
@dbyron-sf Thanks for elaborate supporting details.

**Pull request merged**